src/bin/pgcopydb/ld_transform.c

```diff
@@ -43,11 +44,44 @@ typedef struct TransformStreamCtx
 	uint64_t currentMsgIndex;
 } TransformStreamCtx;
 
+static bool stream_transform_stream_internal(StreamSpecs *specs);
+
+static bool stream_transform_from_queue_internal(StreamSpecs *specs);
+
 static bool canCoalesceLogicalTransactionStatement(LogicalTransaction *txn,
 												   LogicalTransactionStatement *new);
 static bool coalesceLogicalTransactionStatement(LogicalTransaction *txn,
 												LogicalTransactionStatement *new);
 
+/*
+ * stream_transform_init_context_pgsql initializes StreamContext's
+ * transformPGSQL and opens a connection to the target. This is required to use
+ * PQescapeIdentifier API of libpq when escaping identifiers
+ */
+bool stream_transform_init_context_pgsql(StreamSpecs *specs)
+{
+	StreamContext *privateContext = &(specs->private);
+
+	privateContext->transformPGSQL = &(specs->transformPGSQL);
+
+	/* initialize our connection to the target database */
+	if (!pgsql_init(privateContext->transformPGSQL,
+					specs->connStrings->target_pguri,
+					PGSQL_CONN_TARGET))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	if (!pgsql_begin(privateContext->transformPGSQL))
+	{
+		/* errors have already been logged */
+		return false;
+	}
```

Comment:
Why do we need to open a (long-running? idle?) transaction here? It needs to be properly commented/documented, or better yet, avoided if that's possible...

Reply:
Yeah while developing I did this to save 10 seconds of moving `pgsql_open_connection` as a library function. Fixing this.

Reply:
I see. Well please expose something else, a kind of a new wrapper to it, I think we should not expose `pgsql_open_connection` to the outside of the module, because we want to hide some of our PGResults smarts in there and automate closing the connection after the query is complete.

Reply:
Thinking more about it, ending up with an alias for the same function is not going to do any good to the project. Let's make progress with exposing it and maybe revisit later.